var/spack/repos/builtin/packages/rocm-validation-suite/package.py

```diff
@@ -62,9 +63,21 @@ def setup_build_environment(self, build_env):
         depends_on('rocblas@' + ver, when='@' + ver)
         depends_on('rocm-smi-lib@' + ver, when='@' + ver)
 
+    def patch(self):
+        if '@4.5.0:' in self.spec:
+            filter_file('@ROCM_PATH@/rvs',
+                        self.spec.prefix.rvs,
+                        'rvs/conf/deviceid.sh.in',
+                        string=True)
+
     def cmake_args(self):
         args = [
-            self.define('HIP_INC_DIR', self.spec['hip'].prefix),
+            self.define('HIP_DIR', self.spec['hip'].prefix),
+            self.define('HIP_INC_DIR', self.spec['hip'].prefix.include),
+            self.define('HIP_LIB_DIR', self.spec['hip'].prefix.lib),
+            self.define('HSA_INC_DIR', self.spec['hsa-rocr-dev'].prefix.include),
+            self.define('HSA_LIB_DIR', self.spec['hsa-rocr-dev'].prefix.lib),
```

Comment:
i think you can just define HIP_DIR and derive other too accordingly. also the same with HSA_INC_DIR and HSA_LIB_DIR. Instead of single defines for HSA_ROCR_DIR etc.

Reply:
@srekolam ,
Replaced as below in the CMakeList and recipe to reduce redundant env variables.
HIP_DIR with HIP_PATH
HIP_INC_DIR with HIP_PATH/include/
HIP_LIB_DIR with HIP_PATH/lib/
and
HSA_DIR with HSA_PATH
HSA_INC_DIR with HSA_PATH/include/
HIP_LIB_DIR with HSA_PATH/lib/